### PR TITLE
Add system information vars during CMake time, including install method

### PR DESCRIPTION
**Pull request by amytai**
Add some system information during CMake time, including install method.

### CODE REVIEW

File: src/version.h.in [hunk 1]
```diff
@@ -7,5 +7,11 @@
 #define TIMESCALEDB_MINOR_VERSION "@PROJECT_VERSION_MINOR@"
 #define TIMESCALEDB_PATCH_VERSION "@PROJECT_VERSION_PATCH@"
 #define TIMESCALEDB_MOD_VERSION "@VERSION_MOD@"
+#define OS_NAME "@CMAKE_SYSTEM_NAME@"
```
Comment (erimatnor) [inline src/version.h.in]: AFAIK, this is only setting the `OS_NAME` and `OS_VERSION` the source was compiled on. There's no guarantee that these line up with systems the binary runs on, in particular the `OS_VERSION`. Not saying that this is useless info, but not sure it gives the thing you want. Also, perhaps this should be `BUILD_OS_NAME` and `BUILD_OS_VERSION` for clarity?

Comment (amytai) [reply src/version.h.in]: done

**Comment by erimatnor** [comment]
@amytai I still don't see a description in the commit. Is that intentional?

**Comment by amytai** [comment]
Did I fix it..? Are you talking about the PR description?

File: CMakeLists.txt [hunk 2]
```diff
@@ -19,6 +19,8 @@ else ()
   set(PROJECT_VERSION_MOD ${VERSION})
 endif ()
 
+set(PROJECT_INSTALL_METHOD source CACHE STRING "Specify what install platform this binary built is for")
```
Comment (erimatnor) [inline CMakeLists.txt]: built -> build?

Comment (erimatnor) [reply CMakeLists.txt]: or, "...this binary is built for"?

Comment (amytai) [reply CMakeLists.txt]: fixed

**Pull request merged**